examples/cloud-composer-datafusion-ingestion-templates/composer/dag-dependencies/update_metadata.py

```diff
@@ -0,0 +1,373 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module has logic to update coliumn
+descriptions in the data lake tables.
+It reads the data dictionary
+from files and appplies it to BQ tables.
+
+function clean_column_name: cleans any column names
+    that are not BQ compatible to make it compatible
+function derive_metadata_filename: derives the name of the
+    data dictionary file
+function get_metadata_file: reads the contents of
+    the data dictionary file
+function apply_metadata_from_file: patches metadata from
+    the data dictionary file to BQ table columns
+function metadata_update: retrieves the list of all tables
+    populated based on source files and
+    calls apply_metadata_from_file to apply data dictionary
+
+"""
+import csv
+import json
+import logging
+import re
+import time
+import os
+from google.cloud import bigquery
+from google.api_core.exceptions import NotFound
+from airflow.contrib.hooks.gcs_hook import GoogleCloudStorageHook
+from hooks.custom_gcs_hook import GoogleCloudStorageCustomHook
+# class MetadataUpdate:
+
+etl_audit_cols = [
+    "is_active",
+    "record_start_ts",
+    "record_end_ts",
+    "record_ins_ts",
+    "record_upd_ts"
+]
+
+def clean_column_name(col_name):
+    """
+    This function cleans the column name read from
+    the metadata file to conform the column naming standards in BQ.
+    The logic in this function is identical to that used by
+    Data Fusion Autoloader custom plugin to clean column names.
+
+    param col_name: columns name
+    returns (string) cleansed column name
+    """
+
+    col_prefix = "C_"
+    if col_name is None or col_name=="":
+        logging.debug("Column name is null or blank")
+        return col_name
+    else:
+        cleaned_col_name = re.sub("[^0-9a-zA-Z_]+", "", col_name)
+
+        if not cleaned_col_name[0].isalpha():
+            cleaned_col_name = col_prefix + cleaned_col_name
+        if col_name  != cleaned_col_name:
+            logging.debug(
+                f"Cleaned column name from "
+                f"{col_name} to {cleaned_col_name}")
+        return cleaned_col_name
+
+def derive_metadata_filename(
+    system_name,src_file_name,source_regex=None):
+    """
+    Derives the name of the metadata file
+    system_name (string): source system name
+    src_file_name (string): name of souce data file
+    source_regex (string): regex pattern based on which
+        data dictionary file name will be derived
+    returns (string) metadata file name corresponding to the file
+    """
+    src_file_nm = src_file_name.split("/")[-1] # Split the path
+    logging.debug(f"metadata file: {src_file_nm}")
+    if source_regex is not None and source_regex != "":
```

Comment:
this is unnecisarily complex / hard to read.
both `None` and `''` coerce to False
```suggestion
    if source_regex:
```